src/parsers/strategy_file.rs

```diff
@@ -43,16 +44,55 @@ pub fn write(file_name: &str, mut new_file_contents: Vec<StrategyInFile>) -> std
 fn sort_columns(s: &mut StrategyInFile) {
     s.columns.sort_by(|a, b| a.name.cmp(&b.name))
 }
-pub fn to_csv(strategy_file: &str, csv_output_file: &str) -> std::io::Result<()> {
+pub fn to_csv(
+    strategy_file: &str,
+    csv_output_file: &str,
+    custom_classifications: ClassificationConfig,
+) -> std::io::Result<()> {
     let strategies = read(strategy_file)?;
     let p: Vec<String> = strategies
         .iter()
         .flat_map(|strategy| {
             strategy.columns.iter().filter_map(|column| {
-                if column.data_category != DataCategory::General {
+                // We want to output rows that are not 'General' or if they are custom and invalid.
+                let mut include_in_csv = false;
+                let mut validity_info = String::new();
+
+                match &column.data_category {
+                    DataCategory::General => {
+                        // Generally, don't include General unless it's an invalid custom one (edge case, see below)
+                    }
+                    DataCategory::Custom(category_name) => {
+                        if !custom_classifications.is_valid_classification(category_name) {
+                            include_in_csv = true; // Include invalid custom categories
+                            validity_info =
+                                format!(", [INVALID CUSTOM CLASSIFICATION: {}]", category_name);
+                        } else {
+                            include_in_csv = true; // Include valid custom, non-General categories
+                            validity_info =
+                                format!(", [VALID CUSTOM CLASSIFICATION: {}]", category_name);
+                        }
+                    }
+                    // For other built-in, non-General categories (Pii, Unknown, etc.)
+                    _ => {
+                        include_in_csv = true;
+                        // No specific validity_info needed for these built-ins in the CSV as they aren't custom
+                    }
+                }
```

Comment:
We could reduce the need for variable mutability and reduce lines of code a little by directly assigning the results of the `match` expression.

What do you think?

```suggestion
let (include_in_csv, validity_info) = match &column.data_category {
    // Generally, don't include General unless it's an invalid custom one (edge case, see below)
    DataCategory::General => (false, String::new()),

    DataCategory::Custom(category_name) => {
        if !custom_classifications.is_valid_classification(category_name) {
            // Include invalid custom categories
            (
                true,
                format!(", [INVALID CUSTOM CLASSIFICATION: {}]", category_name),
            )
        } else {
            // Include valid custom, non-General categories
            (
                true,
                format!(", [VALID CUSTOM CLASSIFICATION: {}]", category_name),
            )
        }
    }

    // For other built-in, non-General categories (Pii, Unknown, etc.)
    _ => {
        // No specific validity_info needed for these built-ins in the CSV as they aren't custom
        (true, String::new())
    }
};
```